R/view_config_val_errors.R

```diff
@@ -80,10 +80,20 @@ clean_error_df <- function(errors_tbl) {
   if (is.null(errors_tbl)) {
     return(NULL)
   }
+
+  # Move any custom error messages to the message column
+  if (!is.null(errors_tbl$parentSchema$errorMessage)) {
+    error_msg <- !is.na(errors_tbl$parentSchema$errorMessage)
+    errors_tbl$message[error_msg] <- errors_tbl$parentSchema$errorMessage[error_msg]
+  }
+
   errors_tbl[c("dataPath", "parentSchema")] <- NULL
   errors_tbl <- errors_tbl[!grepl("oneOf.+", errors_tbl$schemaPath), ]
+  # remove superfluous if error. The "then" error is what we are interested in
```

Comment:
This is a result of schema validation where two errors are reported when a `then` statement fails. Specifically it's due to how JSON Schema interprets the `if` and `then` clauses. When a `then` statement fails, the associated `if` statement is also considered a failure so we get an error message for both. As we are only interested in the `then` statement, we remove the unhelpful `#if` error row. Here's the relevant section in the schema in version v4.0.1 https://github.com/hubverse-org/schemas/blob/ea42005f0cbccc7367375a2a1debc2ef5996f0ad/v4.0.1/tasks-schema.json#L1493-L1507